### PR TITLE
feature: making search proxy central point for search

### PR DESCRIPTION
**Pull request by gabrielfs7**
Centralize search to proxy, so we can manage the feature flag in a single place

https://oat-sa.atlassian.net/browse/ADF-258

Related PRs

- [x] https://github.com/oat-sa/extension-tao-advanced-search/pull/27
- [x] https://github.com/oat-sa/lib-tao-elasticsearch/pull/45
- [x] https://github.com/oat-sa/extension-tao-publishing/pull/111

### CODE REVIEW

**Review by boajer — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Review by chinnu-focaloid — APPROVED** [review]
- [X] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [X] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: migrations/Version202106010945122237_tao.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+final class Version202106010945122237_tao extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return 'Register ' . SearchProxy::class;
+    }
+
+    public function up(Schema $schema): void
+    {
+        /** @var Search $legacySearch */
```
[the remark this answers is not in the document]
Comment (gabrielfs7) [reply migrations/Version202106010945122237_tao.php:22]: Thanks!

File: migrations/Version202106010945122237_tao.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+        $this->getServiceManager()->register(SearchProxy::SERVICE_ID, $searchProxy);
+    }
+
+    public function down(Schema $schema): void
+    {
+        /** @var SearchProxy $searchProxy */
+        $searchProxy = $this->getServiceManager()->get(SearchProxy::SERVICE_ID);
+
+        /** @var Search $search */
```
[the remark this answers is not in the document]
Comment (gabrielfs7) [reply migrations/Version202106010945122237_tao.php:42]: Thanks!

**Comment by AntonNeviarouski** [comment]
QA verified on http://aadvanced-search-integration.playground.kitchen.it.taocloud.org:48321/

**Pull request merged**